eip-2470.md

```diff
@@ -49,23 +70,47 @@ contract SingletonFactory {
      */
     function deploy(bytes memory _initCode, bytes32 _salt)
         public
+        payable
         returns (address payable createdContract)
     {
         assembly {
-            createdContract := create2(0, add(_initCode, 0x20), mload(_initCode), _salt)
+            createdContract := create2(callvalue(), add(_initCode, 0x20), mload(_initCode), _salt)
         }
+        require(createdContract != address(0), "ERC2470: CREATE2 failed");
+    }
+
+    /**
+     * @notice Calculates the resulting address of a contract created in this factory.
+     * @param _initCode Initialization code.
+     * @param _salt Arbitrary value to modify resulting address.
+     * @return resultAddress predicted address.
+     */
+    function predictedAddress(
+        bytes memory _initCode,
+        bytes32 _salt
+    ) 
+        external 
+        view 
+        returns(address payable resultAddress) 
+    {
+        return address(uint256(keccak256(abi.encodePacked(
+            byte(0xff),
+            address(this),
+            _salt,
+            keccak256(_initCode)
+        ))));
     }
 }
 // IV is a value changed to generate the vanity address.
-// IV: 6583047
+// IV: TBD
```

Comment:
This is already defined in the source above

Reply:
Yeah, this needs to be mined. The rules for mining is "starting with 2470, or have more than 3 zero bytes.

I can handle the deploy procedure and IV number, factory address, and everything else will change. A change in a commented line changes everything.